source/Calamari/Integration/FileSystem/CalamariPhysicalFileSystem.cs

```diff
@@ -152,6 +153,27 @@ public void DeleteDirectory(string path, FailureOptions options)
             }
         }
 
+        static void EnsureDirectoryDeleted(string path, FailureOptions failureOptions)
+        {
+            var retry = GetRetryTracker(); 
+
+            while (retry.Try())
+            {
+                if (!Directory.Exists(path))
+                    return;
+
+                if (retry.CanRetry() && retry.ShouldLogWarning())
+                    Log.VerboseFormat("Waiting for directory '{0}' to be deleted", path);
+            }
+
+            var message = $"Unable to ensure directory '{path}' was deleted";
```

Comment:
I'm not sure we were "unable" to, it's probably more a case of "The directory '{path}' was not deleted". What do you think?